app/controllers/component-tree.js

```diff
@@ -0,0 +1,281 @@
+import EmberObject, {
+  get,
+  computed
+} from '@ember/object';
+import Controller, {
+  inject as controller
+} from '@ember/controller';
+import searchMatch from 'ember-inspector/utils/search-match';
+import {
+  notEmpty,
+  reads
+} from '@ember/object/computed';
+import {
+  isEmpty
+} from '@ember/utils';
+
+/**
+ * ComponentViewItem is used to represent the flattened nodes in the component tree
+ */
+let ComponentViewItem = EmberObject.extend({
```

Comment:
Should we maybe have a separate file for `ComponentViewItem` and import it? This file is a bit long, and might help with separation of concerns if we split the component item and the controller into separate files.